datafeeder/src/main/java/org/georchestra/datafeeder/batch/service/PublishingBatchService.java

```diff
@@ -239,6 +253,11 @@ private void doOnEachRunningDataset(DataUploadJob job, Consumer<DatasetUploadSta
         }
     }
 
+    @VisibleForTesting
+    String formatGsClientServerExceptionMessage(String message) {
+        return message.lines().limit(2).collect(Collectors.joining("\n"));
```

Comment:
And if we want to be a little bit more performant we can use `reduce` instead. 😄 
```suggestion
        return message.lines().limit(2).reduce("", String::concat));
```

Reply:
what does it change ? I feel a bit lost with Java streams to be honest

Reply:
As I understand `reduce` apply modification on stream elements themselves instead of `collect` which execute a function on mutable objects.

I just tried to understand a little bit more and eneded up with this : 
```
		long startTime;
        long stopTime;

        startTime = System.nanoTime();
        System.out.println(error.lines().limit(2).collect(Collectors.joining()));
        stopTime = System.nanoTime();
        System.out.println(stopTime - startTime);

        startTime = System.nanoTime();
        System.out.println(error.lines().limit(2).reduce("", String::concat));
        stopTime = System.nanoTime();
        System.out.println(stopTime - startTime);
```
```
 Error publishing the following datasets:  armoires: Status 500:
2473545
 Error publishing the following datasets:  armoires: Status 500:
727366
```

Reply:
it is unclear to me, but reading https://stackoverflow.com/questions/22577197/java-8-streams-collect-vs-reduce, it looks like collect performs better than reduce. I guess for the needed computing time here to concatenate 2 strings, performance is not an issue, but still.

Reply:
with the suggested reduce() version, we lose the line feeds, but I think it is acceptable.